shared/util/typed-connect.js

```diff
@@ -18,3 +18,16 @@ export class TypedConnector<State, Dispatch, OwnProps, Props> {
     )
   }
 }
+
+class _TypedConnectorV2<Props, StateProps, DispatchProps, OwnProps, Dispatch: Function, State> {
```

Comment:
a class is the only way to specify type parameters explicitly as I understand it.